source/transactions/transactions.rst

```diff
@@ -744,24 +753,59 @@ format:
 
     { ok : 1, writeConcernError: {code: <Number>, errmsg : "..."} }
 
-Mongos Pinning for Sharded Transactions
----------------------------------------
+Sharded Transactions
+--------------------
+
+MongoDB 4.2 (maxWireVersion 8) introduces support for sharded transactions.
+Sharded transactions support all of the same features as single replica set
+transaction but introduce two new driver concepts: mongos pinning and the
+``recoveryToken`` field.
 
-The server supports sharded transactions starting in MongoDB 4.2 (
-maxWireVersion 8). The server requires that drivers MUST send all commands for
-a single transaction to the same mongos.
+Mongos Pinning
+~~~~~~~~~~~~~~
+
+The server requires that drivers MUST send all commands for
+a single transaction to the same mongos (excluding retries of commitTransaction
+and abortTransaction) therefore this spec introduces the concept of pinning a
```

Comment:
I don't think it's supported to send abortTransaction to a different mongos.

It may appear to work because
- if the txnNumber does not match the number on the second mongos, the second mongos will return NoSuchTransaction from beginOrContinue: https://github.com/mongodb/mongo/blob/acddc7f35f0373ccb2e8fe9d45f42304b2b74f95/src/mongo/s/transaction_router.cpp#L514-L519
- if the txnNumber matches the number on the second mongos, the second mongos will return NoSuchTransaction on seeing that it does not have the participant list: https://github.com/mongodb/mongo/blob/acddc7f35f0373ccb2e8fe9d45f42304b2b74f95/src/mongo/s/transaction_router.cpp#L671-L673

Reply:
Yes, I've noted this server behavior elsewhere in the spec but I thought that the server will improve in the future to make aborting on a new mongos possible. The alternative is to not send abort at all (which has the same behavior as sending abort to a new mongos) or remain pinned to a potentially unreachable server and block for 30 seconds in server selection.

Is there a server ticket for this already or should I create one?

Reply:
If the server started supporting aborting a transaction through a recovery mongos, there would probably be drivers changes needed (like passing the participant list back to the client, so the client can pass it to the recovery mongos).

I don't recommend drivers ever sending abortTransaction to a recovery mongos, since we have not tested that against the server and it is currently undefined in the same way using multiple mongos for transaction statements is undefined... But up to you.

Reply:
I opened https://jira.mongodb.org/browse/SERVER-39692 for the server to support (or define) this behavior. Please see this conversation for why we feel drivers must unpin and send abort to a new mongos: https://github.com/mongodb/specifications/pull/471#discussion_r254515917